src/amo/components/AddonReview.js

```diff
@@ -7,43 +12,82 @@ import { setDenormalizedReview, setReview } from 'amo/actions/reviews';
+import log from 'core/logger';
 import OverlayCard from 'ui/components/OverlayCard';
+import type { SetReviewAction, UserReviewType } from 'amo/actions/reviews';
+import type { SubmitReviewParams } from 'amo/api/index';
+import type { ApiStateType } from 'core/reducers/api';
+import type { ErrorHandler as ErrorHandlerType } from 'core/errorHandler';
+import type { ElementEvent } from 'core/types/dom';
+import type { DispatchFunc } from 'core/types/redux';
 
 import 'amo/css/AddonReview.scss';
 
+type AddonReviewProps = {|
+  apiState?: ApiStateType,
+  createLocalStore: typeof defaultLocalStoreCreator,
+  debounce: typeof defaultDebounce,
+  errorHandler: ErrorHandlerType,
+  i18n: Object,
+  onReviewSubmitted: () => Promise<void>,
+  refreshAddon: () => Promise<void>,
+  review: UserReviewType,
+  setDenormalizedReview: (review: $Shape<UserReviewType>) => SetReviewAction,
+  updateReviewText: (review: $Shape<SubmitReviewParams>) => Promise<void>,
+|};
+
+type AddonReviewState = {|
+  reviewBody: ?string,
+|};
 
 export class AddonReviewBase extends React.Component {
-  static propTypes = {
-    apiState: PropTypes.object,
-    errorHandler: PropTypes.object.isRequired,
-    i18n: PropTypes.object.isRequired,
-    onReviewSubmitted: PropTypes.func.isRequired,
-    refreshAddon: PropTypes.func.isRequired,
-    review: PropTypes.object.isRequired,
-    setDenormalizedReview: PropTypes.func.isRequired,
-    updateReviewText: PropTypes.func.isRequired,
-  }
-
-  constructor(props) {
+  localStore: LocalStore;
+  props: AddonReviewProps;
+  reviewForm: Node;
+  reviewPrompt: Node;
+  reviewTextarea: Node;
+  state: AddonReviewState;
+
+  static defaultProps = {
+    createLocalStore: defaultLocalStoreCreator,
+    debounce: defaultDebounce,
+  };
+
+  constructor(props: AddonReviewProps) {
     super(props);
-    this.state = { reviewBody: props.review.body };
-    this.reviewTextarea = null;
+    this.state = {
+      reviewBody: props.review.body,
+    };
+    this.localStore = props.createLocalStore(`AddonReview:${props.review.id}`);
+    this.checkForStoredState();
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentWillReceiveProps(nextProps: AddonReviewProps) {
     const { review } = nextProps;
     if (review) {
       this.setState({ reviewBody: review.body });
     }
   }
 
-  onSubmit = (event) => {
+  checkForStoredState() {
+    return this.localStore.getData()
+      .then((data) => {
+        if (data) {
+          log.debug(oneLine`Initializing AddonReview data from LocalStore
+            ${this.localStore.id}`, data);
+          this.setState(data);
+        }
+      });
+  }
+
+  onSubmit = (event: Event) => {
     const { apiState, errorHandler, onReviewSubmitted, review } = this.props;
     const { reviewBody } = this.state;
     event.preventDefault();
     event.stopPropagation();
 
-    const newReviewParams = { body: reviewBody };
+    const newReviewParams = { body: reviewBody || undefined };
```

Comment:
The review params here are for setting a review object which is typed independently of the local storage thing. If I don't initialize this to `undefined`, Flow says:
````
src/amo/components/AddonReview.js:90
 90:     const newReviewParams = { body: reviewBody };
                                         ^^^^^^^^^^ null. This type is incompatible with
  8:   body?: string,
              ^^^^^^ string. See: src/amo/actions/reviews.js:8

src/amo/components/AddonReview.js:90
 90:     const newReviewParams = { body: reviewBody };
                                         ^^^^^^^^^^ null. This type is incompatible with
 33:   body?: string,
              ^^^^^^ string. See: src/amo/api/index.js:33
````
This is because the [body parameter](https://github.com/kumar303/addons-frontend/blob/4f5ea85cf6cd94d356241d2c5e1ece1c299de77d/src/amo/actions/reviews.js#L8) can be undefined but not `null` or `false` or anything else.